spv/envelope.go

```diff
@@ -36,3 +38,72 @@ func (e *Envelope) ParentTx(txID string) (*bt.Tx, error) {
 	}
 	return bt.NewTxFromString(env.RawTx)
 }
+
+// Bytes takes an spvEnvelope struct and returns the serialised binary format.
+func (e *Envelope) Bytes() ([]byte, error) {
+	flake := make([]byte, 0)
+
+	// Binary format version 1
+	flake = append(flake, 1)
+
+	initialTx := map[string]*Envelope{
+		e.TxID: {
+			TxID:          e.TxID,
+			RawTx:         e.RawTx,
+			Proof:         e.Proof,
+			MapiResponses: e.MapiResponses,
+			Parents:       e.Parents,
+		},
+	}
+
+	err := serialiseInputs(initialTx, &flake, true)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return flake, nil
+}
+
+func serialiseInputs(parents map[string]*Envelope, flake *[]byte, root bool) error {
```

Comment:
What do you think about changing this to `serialiseParents` or just `serialise` or something just to distinguish it from the other main serialiseInputs func for tx inputs (I know it's a private func but just being a bit pedantic since this will be reference code for others to use)

Reply:
Good idea, done.